I encountered a panic when running `make testplannerlogic TESTCONFIG=local` with "streaming" by default
```
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range [recovered]
	panic: panic while executing 1 statements: SELECT * FROM _@_; caused by runtime error: index out of range

goroutine 9262 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc00a5d6000, 0x72a5ec0, 0xc00577db80, 0x680e400, 0x8d4a210)
	/Users/yahoryuzefovich/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:727 +0x330
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc00a5d6000, 0x72a5ec0, 0xc00577db80)
	/Users/yahoryuzefovich/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:440 +0x61
panic(0x680e400, 0x8d4a210)
	/usr/local/opt/go/libexec/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/sql/exec.CatchVectorizedRuntimeError.func1(0xc0066cc1c8)
	/Users/yahoryuzefovich/go/src/github.com/cockroachdb/cockroach/pkg/sql/exec/error.go:67 +0x5d8
panic(0x680e400, 0x8d4a210)
	/usr/local/opt/go/libexec/src/runtime/panic.go:522 +0x1b5
github.com/cockroachdb/cockroach/pkg/sql/row.(*CFetcher).processValue(0xc00837a000, 0x72a5f80, 0xc00a22e240, 0x0, 0xc007e35800, 0xc008572178, 0xc009a3e028, 0x1, 0x1, 0xc007d85680)
	/Users/yahoryuzefovich/go/src/github.com/cockroachdb/cockroach/pkg/sql/row/cfetcher.go:788 +0x1335
github.com/cockroachdb/cockroach/pkg/sql/row.(*CFetcher).NextBatch(0xc00837a000, 0x72a5f80, 0xc00a22e240, 0xc007df0000, 0xc0066cbbc0, 0x5389bde, 0x0)
	/Users/yahoryuzefovich/go/src/github.com/cockroachdb/cockroach/pkg/sql/row/cfetcher.go:629 +0x65b
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*colBatchScan).Next(0xc009afa9c0, 0x72a5f80, 0xc00a22e240, 0xed4c1aa84, 0x0)
	/Users/yahoryuzefovich/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/colbatch_scan.go:49 +0x47
github.com/cockroachdb/cockroach/pkg/sql/exec.(*CancelChecker).Next(0xc007e3b440, 0x72a5f80, 0xc00a22e240, 0x4, 0xd681890)
	/Users/yahoryuzefovich/go/src/github.com/cockroachdb/cockroach/pkg/sql/exec/cancel_checker.go:40 +0x69
github.com/cockroachdb/cockroach/pkg/sql/exec.(*VectorizedStatsCollector).Next(0xc00240d590, 0x72a5f80, 0xc00a22e240, 0x0, 0x0)
	/Users/yahoryuzefovich/go/src/github.com/cockroachdb/cockroach/pkg/sql/exec/stats.go:82 +0x7b
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*materializer).next(0xc002bc2300, 0x402e71f, 0x8, 0xc00642f300, 0x59e7a89)
	/Users/yahoryuzefovich/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/materializer.go:116 +0x451
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*materializer).nextAdapter(...)
	/Users/yahoryuzefovich/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/materializer.go:107
github.com/cockroachdb/cockroach/pkg/sql/exec.CatchVectorizedRuntimeError(0xc0066cc1d8, 0x0, 0x0)
```
and I think we want to catch such panics and turn them into errors in the future.

cc @rohany for the cause

--- a/pkg/sql/exec/error.go
+++ b/pkg/sql/exec/error.go
@@ -80,6 +80,8 @@ func CatchVectorizedRuntimeError(operation func()) (retErr error) {
 const (
 	execPackagePrefix  = "github.com/cockroachdb/cockroach/pkg/sql/exec"
 	colBatchScanPrefix = "github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*colBatchScan)"
+	cFetcherPrefix     = "github.com/cockroachdb/cockroach/pkg/sql/row.(*CFetcher)"
+	columnarizerPrefix = "github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*columnarizer)"
 )
 
 // isPanicFromVectorizedEngine checks whether the panic that was emitted from
@@ -88,7 +90,9 @@ const (
 // panicEmittedFrom must be trimmed to not have any white spaces in the prefix.
 func isPanicFromVectorizedEngine(panicEmittedFrom string) bool {
 	return strings.HasPrefix(panicEmittedFrom, execPackagePrefix) ||
-		strings.HasPrefix(panicEmittedFrom, colBatchScanPrefix)
+		strings.HasPrefix(panicEmittedFrom, colBatchScanPrefix) ||
+		strings.HasPrefix(panicEmittedFrom, cFetcherPrefix) ||
+		strings.HasPrefix(panicEmittedFrom, columnarizerPrefix)
 }
 
 // TestVectorizedErrorEmitter is an Operator that panics on every odd-numbered

